### PR TITLE
Only show core navigation elements (call/chat/notification/info) when a widget is maximised

### PR DESCRIPTION
**Pull request by toger5**
Fixes: https://github.com/vector-im/element-web/issues/19632
image

---
Here's what your changelog entry will look like:

## ✨ Features
 * Only show core navigation elements (call/chat/notification/info) when a widget is maximised ([\#7114](https://github.com/matrix-org/matrix-react-sdk/pull/7114)). Fixes vector-im/element-web#19632 and vector-im/element-web#19632. Contributed by @toger5.

Preview: https://61a50108611fbe36f7a96ef1--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/rooms/RoomHeader.tsx [hunk 1]
```diff
@@ -57,6 +57,7 @@ interface IProps {
     e2eStatus: E2EStatus;
     appsShown: boolean;
     searchInfo: ISearchInfo;
+    coreElementsOnly?: boolean;
```
Comment (turt2live) [inline src/components/views/rooms/RoomHeader.tsx:60]: can we get a line or two of comments on this to explain what core elements are please?

**Review by turt2live — CHANGES_REQUESTED** [review]
otherwise lgtm

**Review by gaelledel — APPROVED** [review]

File: src/components/views/right_panel/RoomHeaderButtons.tsx [hunk 2]
```diff
@@ -158,18 +159,18 @@ export default class RoomHeaderButtons extends HeaderButtons<IProps> {
...
         return <>
-            <PinnedMessagesHeaderButton
+            { !this.props.coreElementsOnly && <PinnedMessagesHeaderButton
                 room={this.props.room}
                 isHighlighted={this.isPhase(RightPanelPhases.PinnedMessages)}
                 onClick={this.onPinnedMessagesClicked}
-            />
-            <TimelineCardHeaderButton
+            /> }
+            { this.props.coreElementsOnly && <TimelineCardHeaderButton
```
Comment (jryans) [inline src/components/views/right_panel/RoomHeaderButtons.tsx:167]: It's a bit awkward to use `coreElementsOnly` as the signal to **both** filter out non-core and **also add** the timeline for maximised widgets. Maybe this part here takes a separate prop about whether there's a maximised widget displayed? That would make the logic more obvious.

Comment (toger5) [reply src/components/views/right_panel/RoomHeaderButtons.tsx:167]: Just to clarify:
Two props: `coreElementOnly` + `showTimelineCardHeaderButton`
Or one porp like now but renamed to sth like: `nonTimelineMainSplit`

File: src/components/structures/RoomView.tsx [hunk 3]
```diff
@@ -2155,7 +2155,21 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
             // TODO-video MainSplitContentType.Video:
             //     break;
         }
-
+        let excludedRighPanelPhaseButtons = [RightPanelPhases.Timeline];
```
Comment (jryans) [inline src/components/structures/RoomView.tsx:2158]:
```suggestion
        let excludedRightPanelPhaseButtons = [RightPanelPhases.Timeline];
```

**Review by jryans — CHANGES_REQUESTED** [review]
A small typo, but otherwise looks good to me! 😄

File: src/components/views/right_panel/RoomHeaderButtons.tsx (hunk 2 above)
Comment (jryans) [reply src/components/views/right_panel/RoomHeaderButtons.tsx]: Seems like you have found a good way with the excluded set. 😄

**Review by jryans — APPROVED** [review]

**Pull request merged**